src/app/core/testing/mock-enclosure/mock-storage-generator.utils.ts

```diff
@@ -1,984 +1,54 @@
-        for (let i = 0; i < width; i++) {
-          const topologyDisk: TopologyDisk = this.generateTopologyDisk();
-          vdevDisks.push(topologyDisk);
-        }
-
-        const vdev = {
-          type: layout,
-          children: vdevDisks,
-          guid: Number(12345).toString(),
-          unavail_disk: null,
-          stats: { size: null } as TopologyItemStats,
-          status: TopologyItemStatus.Online,
-        } as VDev;
-
-        if (vdevName) vdev.name = vdevName;
-
-        return vdev;
-      }
-      default:
-        throw new Error('Invalid TopologyItemType. Please use STRIPE, MIRROR, RAIDZ etc');
-    }
+  webuiDashboardEnclosureResponse(): ApiCallResponse<'webui.enclosure.dashboard'> {
+    return this.enclosures;
   }
 
-  private generateTopologyDisk(): TopologyDisk {
+  enhanceSystemInfoResponse(response: SystemInfo): SystemInfo {
     return {
-      type: TopologyItemType.Disk,
-      status: TopologyItemStatus.Online,
-      children: [],
-      disk: null,
-      stats: { size: null } as TopologyItemStats,
-      device: null,
-      guid: Number(12345).toString(),
-      unavail_disk: null,
-    } as TopologyDisk;
-  }
-
-  private calculateVdevCapacity(vdev: VDev): number {
-    const sizes: number[] = vdev.children.map((child) => child.stats.size);
-    const uniqueDiskSizes = new Set(sizes);
-    const smallestSize: number = Math.min.apply(this, [...uniqueDiskSizes]);
-
-    switch (vdev.type) {
-      case TopologyItemType.Mirror:
-        return vdev.children[0].stats.size;
-      case TopologyItemType.Raidz:
-      case TopologyItemType.Raidz1:
-        return smallestSize * (sizes.length - 1);
-      case TopologyItemType.Raidz2:
-        return smallestSize * (sizes.length - 2);
-      case TopologyItemType.Raidz3:
-        return smallestSize * (sizes.length - 3);
-      default:
-        return undefined;
-    }
-  }
-
-  // Generate Disks
-  private getMinLayoutWidth(layout: TopologyItemType): number {
-    let width = 0;
-
-    switch (layout) {
-      case TopologyItemType.Stripe:
-        width = 1;
-        break;
-      case TopologyItemType.Mirror:
-        width = 2;
-        break;
-      case TopologyItemType.Raidz1:
-        width = 3;
-        break;
-      case TopologyItemType.Raidz2:
-        width = 4;
-        break;
-      case TopologyItemType.Raidz3:
-        width = 5;
-        break;
-      default:
-        width = 0;
-        break;
-    }
-
-    return width;
-  }
-
-  private generateDiskName(diskCount: number, startIndex = 0, diskType: MockDiskType = MockDiskType.Hdd): string[] {
-    const diskNames: string[] = [];
-
-    const generateName = (index: number): string => {
-      let result = '';
-      do {
-        result = (index % 26 + 10).toString(36) + result;
-        index = Math.floor(index / 26) - 1;
-      } while (index >= 0);
-      return result;
+      ...response,
+      system_product: this.config.controllerModel,
+      system_manufacturer: 'iXsystems',
```

Comment:
```suggestion
      ...response,
      platform: `TRUENAS-${this.config.controllerModel}`,
      system_product: `TRUENAS-${this.config.controllerModel}`,
      remote_info: {
        ...response.remote_info,
        platform: `TRUENAS-${this.config.controllerModel}`,
        system_product: `TRUENAS-${this.config.controllerModel}`,
      },
```
`TRUENAS-` prefix is used to determine enclosure in system features.
`system_product` called `platform` in `webui.main.dashboard.sys_info`, so needs both
`remote_info` is system info for standby node in `webui.main.dashboard.sys_info`
`truenas.is_ix_hardware` is used to determine the manufacturer
`system_manufacturer` is not used anymore

Reply:
I only see one check for `TRUENAS` in system product. I don't think we should rely on it.
`truenas.is_ix_hardware` is already mocked in `MockEnclosureWebsocketService`.
Added remote_info and platform.

Reply:
System Info widgets rely on the `selectEnclosureSupport` selector, whose value depends on the prefix in `system_product`. This selector's source could be changed, but otherwise, some functions do not work correctly.

Reply:
Yeah, I saw it, I think we'd need to update it.